lib/screen/workout/workout_list.dart

```diff
@@ -0,0 +1,198 @@
+import 'package:event_bloc/event_bloc.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:simple_exercise_builder/bloc/workout/workout.dart';
+import 'package:simple_exercise_builder/model/workout.dart';
+import 'package:simple_exercise_builder/screen/workout/search_bar.dart';
+import 'package:simple_exercise_builder/screen/workout/workout_category_list.dart';
+
+class WorkoutList extends StatefulWidget {
+  final void Function(Workout exercise)? action;
+  final bool enableAdding;
+  final bool enableModifying;
+
+  const WorkoutList({
+    Key? key,
+    this.action,
+    this.enableAdding = true,
+    this.enableModifying = true,
+  }) : super(key: key);
+
+  @override
+  _WorkoutListState createState() => _WorkoutListState();
+}
+
+class _WorkoutListState extends State<WorkoutList> {
+  bool editMode = false;
+
+  @override
+  Widget build(BuildContext context) {
+    final bloc = BlocProvider.watch<WorkoutBloc>(context);
+    final workoutMap = bloc.workoutMap;
+    final workoutList = bloc.workoutList
+        .map((workoutID) => workoutMap[workoutID])
+        .where((workout) => workout != null);
+    final userMadeWorkouts = workoutList.where((workout) => workout!.userMade);
+    final preMadeWorkouts = workoutList.where((workout) => !workout!.userMade);
```

Comment:
Instead of the above, you can just have the WorkoutBloc give you this (still as a List of ints). This is a bit too much work to be doing in the UI layer.